apps/aecore/src/aec_peer_connection.erl

```diff
@@ -222,6 +219,14 @@ handle_cast({send_block, Hash}, State) ->
 handle_cast(stop, State) ->
     State1 = cleanup_connection(State),
     {stop, normal, State1};
+handle_cast(disconnect, #{ status := {connecting, _} } = S) ->
+    epoch_sync:debug("Disconnect received, aborting connection"),
+    {stop, normal, S};
+handle_cast(disconnect, #{ status := {connected, _} } = S) ->
+    epoch_sync:debug("Disconnect received, closing connection"),
+    {noreply, close(S)};
+handle_cast(disconnect, #{ status := {disconnecting, _} } = S) ->
```

Comment:
I see this commit introduces status`disconnecting` (connecting, connected, disconnected were already present). Ok.

Reply:
Is it a problem? What do you mean by this comment?

Reply:
This was added to support clean disconnection through the new `close` message.